helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/SkipDecoder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.compatibility;
+
+import java.io.IOException;
+import org.apache.avro.Schema;
+import org.apache.avro.io.Decoder;
+
+/**
+ * An abstract class that extends from Decoder that supports skipping various
+ * symbol types and get the variable length of the fields.
+ */
+public abstract class SkipDecoder extends Decoder {
+
+  public abstract int readStringSize() throws IOException;
```

Comment:
needs doc - does this skip the  underlying input? does ti retrn the size of the fiels skipped? is that size in bytes? even for json? does it include the  size of payload alone, or also the size of the int  used to describe payload length?

Reply:
similarly for all methods here